pallets/passkey/src/lib.rs

```diff
@@ -337,38 +353,72 @@ where
 		let runtime_call: <T as frame_system::Config>::RuntimeCall =
 			<T as frame_system::Config>::RuntimeCall::from(self.1.clone());
 		let who = self.0.clone();
-		let fee = self.withdraw_token_fee(&who, &runtime_call, info, len, Zero::zero())?;
 
-		let priority = pallet_transaction_payment::ChargeTransactionPayment::<T>::get_priority(
+		pallet_transaction_payment::ChargeTransactionPayment::<T>::from(Zero::zero()).validate(
+			&who,
+			&runtime_call,
 			info,
 			len,
-			Zero::zero(),
-			fee,
-		);
+		)
+	}
+}
+
+/// Frame system related checks
+#[derive(Encode, Decode, Clone, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct FrameSystemChecks<T: Config + Send + Sync>(pub T::AccountId, pub Call<T>);
+
+impl<T: Config + Send + Sync> FrameSystemChecks<T>
+where
+	<T as frame_system::Config>::RuntimeCall:
+		From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+{
+	/// Validates the transaction fee paid with tokens.
+	pub fn pre_dispatch(&self) -> Result<(), TransactionValidityError> {
+		let info = &self.1.get_dispatch_info();
+		let len = self.0.using_encoded(|c| c.len());
+		let runtime_call: <T as frame_system::Config>::RuntimeCall =
+			<T as frame_system::Config>::RuntimeCall::from(self.1.clone());
+		let who = self.0.clone();
 
-		Ok(ValidTransaction { priority, ..Default::default() })
+		let non_zero_sender_check = frame_system::CheckNonZeroSender::<T>::new();
+		let spec_version_check = frame_system::CheckSpecVersion::<T>::new();
+		let tx_version_check = frame_system::CheckTxVersion::<T>::new();
+		let genesis_hash_check = frame_system::CheckGenesis::<T>::new();
+		let era_check = frame_system::CheckEra::<T>::from(Era::immortal());
```

Comment:
Order of check is kept same as in [SignedExtras](https://github.com/frequency-chain/frequency/blob/d25b833d77a9968fa5d01d7e21e948e4d0ee16d9/runtime/frequency/src/lib.rs#L260)